hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -140,9 +140,16 @@ public static Schema readAvroSchema(Configuration configuration, Path parquetFil
    * Read out the bloom filter from the parquet file meta data.
    */
   public static BloomFilter readBloomFilterFromParquetMetadata(Configuration configuration, Path parquetFilePath) {
-    String footerVal =
-        readParquetFooter(configuration, parquetFilePath, HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY)
-            .get(0);
+    List<String> footerVals =
+        readParquetFooter(configuration, parquetFilePath, HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY,
+            HoodieAvroWriteSupport.OLD_HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY);
+    String footerVal;
+    if (null != footerVals.get(0)) {
```

Comment:
ah you have it here already.. anyways, so the list will contain a null of the org.apache.hudi key is not found? Bit strange API.

Reply:
does nt line 128 above throw an error instead? Can we change the return type to map above and clean this up?